util-core/src/main/scala/com/twitter/util/Duration.scala

```diff
@@ -373,16 +373,23 @@ sealed class Duration private[util] (protected val nanos: Long) extends {
     def overflowedDuration(a: Long, b: Long): Duration =
       if ((a < 0) == (b < 0)) Duration.Top else Duration.Bottom
 
-    // a < b
-    def multiplyNanos(a: Long, b: Long): Duration =
-      if (a < 0L)
-        if (b < 0L && a < Long.MaxValue / b) overflowedDuration(a, b)
-        else if (b > 0L && Long.MinValue / b > a) overflowedDuration(a, b)
-        else Duration.fromNanoseconds(a * b)
-      else if (a > 0L && a > Long.MaxValue / b) overflowedDuration(a, b)
-      else Duration.fromNanoseconds(a * b)
-
-    if (nanos > x) multiplyNanos(x, nanos) else multiplyNanos(nanos, x)
+    def multiplyNanos(a: Long, b: Long): Duration = {
```

Comment:
I'm somewhat skeptical this actually improves readability. Perhaps others might find it more readable.

Reply:
I agree that it's not drastic improvement, maybe it reads slightly better? I was also thinking of adding comments at the end of certain lines, as below: 

```
larger match {
  case l if (l == 0 || smaller == 0) => Duration.fromNanoseconds(0)                       // zero
  case l if (l > 0) => smaller match {
    case s if (s > 0 && Long.MaxValue / l >= smaller) => Duration.fromNanoseconds(a * b)  // + times + = + result
    case s if (s < 0 && Long.MinValue / l <= smaller) => Duration.fromNanoseconds(a * b)  // + times - = - result
    case _ => overflowedDuration(a, b)
  }
  case l if (Long.MaxValue / l) <= smaller => Duration.fromNanoseconds(a * b)             // - times - = + result
  case _ => overflowedDuration(a, b)
}
```